proto/teleport/lib/vnet/diag/v1/diag.proto

```diff
@@ -18,8 +18,116 @@ syntax = "proto3";
 
 package teleport.lib.vnet.diag.v1;
 
+import "google/protobuf/timestamp.proto";
+
 option go_package = "github.com/gravitational/teleport/gen/proto/go/teleport/lib/vnet/diag/v1;diagv1";
 
+// Report represents the results of running individual checks. It also includes general information
+// about the network stack managed by VNet. It assumes that each individual check as well as getting
+// info about the network stack can fail.
+message Report {
+  // created_at is the UTC timestamp at which the report was generated.
+  google.protobuf.Timestamp created_at = 1;
+  NetworkStackResult network_stack_result = 2;
+  repeated CheckResult check_results = 3;
+}
+
+// NetworkStackResult represents the attempt at getting information about the network stack managed
+// by VNet.
+message NetworkStackResult {
+  CheckStatus status = 1;
+  // error is present if status is CHECK_STATUS_ERROR.
+  string error = 2;
+  NetworkStack network_stack = 3;
+}
+
+// NetworkStack describes the network stack managed by VNet.
+message NetworkStack {
+  // interface_name is the name of the interface set up and used by VNet.
+  string interface_name = 1;
+  // ipv4_cidr_ranges are all the possible ranges under which VNet is going to assign IPv4 addresses
+  // for apps. The first IP of the first range is used for the TUN device.
+  // Each root cluster can specify its own CIDR range to be used for apps within that profile.
+  repeated string ipv4_cidr_ranges = 2;
+  // ipv6_prefix is the IPv6 prefix under which VNet creates IPv6 addresses for apps and its DNS
+  // server.
+  string ipv6_prefix = 3;
+  // dns_zones lists domains for which DNS queries are supposed to be captured by VNet.
+  repeated string dns_zones = 4;
+}
+
+// CheckResult describes the attempt at running a particular diagnostic check.
+message CheckResult {
+  CheckStatus status = 1;
+  // error is present if status is CHECK_STATUS_ERROR.
+  string error = 2;
+  // check_report is the output data of running a particular check.
+  //
+  // If status is CHECK_STATUS_ERROR, check_report is set to CheckReport with an empty check_report
+  // inside of the correct type. This means that CheckResult can be distinguished from other results
+  // even if the check failed.
```

Comment:
This is my third comment on this wording (I deleted the first two). I think I understand it now, but it took a while.

1. as mentioned in another comment, I think if `check_report` (the oneof) were renamed to `report`, this would read a bit better.
2. I think this needs just a little elaboration, such as "If status is CHECK_STATUS_ERROR (which means the check failed to complete), ..." and extending the end to say "if the check failed to run.". I see below that comment is on the `CheckStatus` enum, but to understand this comment I was bouncing around a lot and I still didn't get it until I read the code for the route status check. This is mostly because the distinction between a check with issues and a check result error was not all that clear initially.

Reply:
Hmm, I think the fact that I use both "result" and "report" to mean completely different things is also confusing. I'm in the process of adding more examples to the comments, but I'll also rename "result" to "attempt".

As in, an attempt at running a check can succeed or fail. If it succeeds, it produces a report which indicates whether the check has found any issues.

Reply:
I allowed myself to rebase proto changes onto individual commits, otherwise this would all be doubly confusing for Nic who would have to understand the terminology before and after changes in protos. GitHub fortunately lets you inspect [the changes to the .proto file itself](https://github.com/gravitational/teleport/compare/62caa030f99dbab9904d02b6b0119c5bd57eaebc..864ddb1e647bd1ffd4a2a1c2cb3d9f07f12016aa#diff-f8e4330260fe79da776adb995b533b8a28284479714aebae149f4365d81ff923) that I made in the force push.